internal/storage/image.go

```diff
@@ -897,6 +900,11 @@ func (st nativeStorageTransport) ResolveReference(ref types.ImageReference) (typ
 	return istorage.ResolveReference(ref)
 }
 
+// UpdatePinnedImagesList updates pinned images list in imageService.
+func (svc *imageService) UpdatePinnedImagesList(configPinnedImage []string) {
+	svc.regexForPinnedImages = append(svc.regexForPinnedImages, CompileRegexpsForPinnedImages(configPinnedImage)...)
```

Comment:
I'm not sure we should append here. Wouldnt' we just want to overwrite the slice? someone could theoretically unpin an image and we'd want it to qualify for GC then